apps/test/unit/lib/levelbuilder/rubrics/RubricsContainerTest.js

```diff
@@ -16,13 +17,86 @@ describe('RubricsContainerTest', () => {
     lessonNumber: 0,
   };
 
+  const rubricInfo = {
+    learningGoals: [
+      {
+        key: 'ui-1',
+        id: 'ui-1',
+        learningGoal: '',
+        aiEnabled: false,
+        position: 1,
+        learningGoalEvidenceLevelsAttributes: [
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.NONE,
+            aiPrompt: '',
+          },
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.LIMITED,
+            aiPrompt: '',
+          },
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.CONVINCING,
+            aiPrompt: '',
+          },
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.EXTENSIVE,
+            aiPrompt: '',
+          },
+        ],
+      },
+      {
+        key: 'ui-2',
+        id: 'ui-2',
+        learningGoal: '',
+        aiEnabled: false,
+        position: 2,
+        learningGoalEvidenceLevelsAttributes: [
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.NONE,
+            aiPrompt: '',
+          },
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.LIMITED,
+            aiPrompt: '',
+          },
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.CONVINCING,
+            aiPrompt: '',
+          },
+          {
+            teacherDescription: '',
+            understanding: RubricUnderstandingLevels.EXTENSIVE,
+            aiPrompt: '',
+          },
+        ],
+      },
+    ],
+  };
+
   it('renders the components on the page correctly for a new rubric', () => {
     const wrapper = mount(<RubricsContainer {...defaultProps} />);
+    expect(wrapper.find('Heading1').text()).to.equal('Create your rubric');
     expect(wrapper.find('select#rubric_level_id option')).to.have.length(3);
     expect(wrapper.find(LearningGoalItem)).to.have.length(1);
     expect(wrapper.find('Button[text="Save your rubric"]')).to.have.length(1);
   });
 
+  it('renders "the components on the page correctly for an exisiting rubric"', () => {
+    const props = {...defaultProps, rubric: rubricInfo};
+    const wrapper = mount(<RubricsContainer {...props} />);
+    expect(wrapper.find('Heading1').text()).to.equal('Modify your rubric');
+    expect(wrapper.find('select#rubric_level_id option')).to.have.length(3);
```

Comment:
It was tricky to immediately understand why the answer to the length of this should be `3`. What is your test philosophy about this situation? Is it possible to consider:

```
expect(wrapper.find('select#rubric_level_id option')).to.have.length(props.levels.length);
```

The above tells me that it should be specifically responding to the `levels` list in the properties.

This way the test is not dependent on the structure of the default properties, but instead flexible to it. If somebody adds or modifies the default properties, the test will still pass (or it will fail because it now covers a different case... in that case: yay! tests working as they are meant to!)

It's probably not too important to be _this_ robust in tests.. or at least these tests, but this style also lets us potentially randomize the test and create a "fuzzy" case, which I'm a big fan of:

```
// Randomly add a few more levels
props.levels.concat(
  [...Array(random(5,8))].map((_, i) => { return {id: i, name: `level ${i}`}; })
);

const wrapper = mount(<RubricsContainer {...props} />);

// Make sure it handles an arbitrary number of levels
expect(wrapper.find('select#rubric_level_id option')).to.have.length(props.levels.length);
```

Though this is a bit out of the ordinary for our codebase and relies on being proactive about tests that may probabilistically fail. (though this type of flaky test is Good because it Caught Something Wrong!)

Reply:
Good points.  

This makes me think of at least one time where I added a test to an existing component which required me to add to existing starting data (such as data about levels in this case).  At times this has caused me to break other tests for a variety of reasons which then caused me to edit those tests too.

I can see how this approach (making tests flexible to changing starting data), could help reduce that from happening down the road if someone else were to add to and/or modify the starting data.  I will make the change!  Thanks for walking me through your thinking!